### PR TITLE
Sanity

### PR DESCRIPTION
**Pull request by PlinHost**
## About The Pull Request
Doc: [link](https://docs.google.com/document/d/1AG7ZsL0MQergzswM9vawFmjNIM3aaEBPVS_PcuohGOU/edit)
Delusion and Signs are disabled as they are not ready yet.

## Changelog
:cl:
add: Added sanity system
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 1]
```diff
@@ -83,4 +83,15 @@
...
+	var/datum/sanity/sanity = null
+
+	var/shock_resist = 0 // Resistance to paincrit
+
+	var/image/fabric_image
+
+	var/language_blackout = 0
+	var/suppress_communication = 0
+
+	var/war_is_hell = 0
```
Comment (LethalGhost) [inline code/modules/mob/living/carbon/human/human_defines.dm]: ???

Comment (PlinHost) [reply code/modules/mob/living/carbon/human/human_defines.dm]: Shit, I forgot to rename it.

File: code/modules/mob/living/carbon/human/say.dm [hunk 2]
```diff
@@ -1,10 +1,61 @@
+/mob/living/carbon/human/proc/suppressed_message()
```
Comment (LethalGhost) [inline code/modules/mob/living/carbon/human/say.dm]: May be it's better to name it something like `get_suppressed_message`?
```suggestion
/mob/living/carbon/human/proc/get_suppressed_message()
```

File: code/modules/sanity/breakdowns.dm [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+
+/datum/breakdown/common/signs/occur()
+	RegisterSignal(holder.owner, COMSIG_HUMAN_SAY, .proc/check_message)
+	return ..()
+
+/datum/breakdown/common/signs/conclude()
+	UnregisterSignal(holder.owner, COMSIG_HUMAN_SAY)
+	..()
+
+/datum/breakdown/common/signs/proc/check_message(_, msg)
```
Comment (LethalGhost) [inline code/modules/sanity/breakdowns.dm]: Is `_` valid arg name?

Comment (PlinHost) [reply code/modules/sanity/breakdowns.dm]: Yes, it works. I use `_` for unused args that are required by signals.

File: code/modules/sanity/sanity_mob.dm [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+	RegisterSignal(owner, COMSIG_HUMAN_SAY, .proc/onSay)
+
+/datum/sanity/proc/onLife()
+	var/affect = SANITY_PASSIVE_GAIN
+	if(owner.stat)
+		changeLevel(affect)
+		return
+	if(!(owner.sdisabilities & BLIND) && !owner.blinded)
+		affect += handle_area()
+		affect -= handle_view()
```
Comment (LethalGhost) [inline code/modules/sanity/sanity_mob.dm]: Shouldn't it be _affect **+=** handle_view()_?

Comment (PlinHost) [reply code/modules/sanity/sanity_mob.dm]: No, `handle_view` returns damage.

**Pull request merged**